frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/__tests__/EventListenerURL.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { shallow, ShallowWrapper } from 'enzyme';
+import EventListenerURL from '../EventListenerURL';
+import { useEventListenerURL } from '../../utils/triggers';
+import {
+  EventlistenerTestData,
+  EventlistenerTypes,
+} from '../../../../test-data/event-listener-data';
+import { ClipboardCopy } from '@patternfly/react-core';
+
+jest.mock('react-i18next', () => {
+  const reactI18next = require.requireActual('react-i18next');
+  return {
+    ...reactI18next,
+    useTranslation: () => ({ t: (key) => key }),
+  };
+});
+
+jest.mock('../../utils/triggers', () => ({ useEventListenerURL: jest.fn<string | null>() }));
+
+type EventListenerURLProps = React.ComponentProps<typeof EventListenerURL>;
+
+describe('EventListenerURL', () => {
+  let wrapper: ShallowWrapper<EventListenerURLProps>;
+  const mockEventListener = EventlistenerTestData[EventlistenerTypes.TRIGGER_REF];
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should return an empty render if a routeURL does not exist', () => {
+    (useEventListenerURL as jest.Mock).mockReturnValueOnce(null);
+    wrapper = shallow(
+      <EventListenerURL eventListener={mockEventListener} namespace="test-namespace" />,
+    );
+    expect(wrapper.isEmptyRender()).toBe(true);
+  });
+
+  it('should return render the URL if a routeURL exists', () => {
+    (useEventListenerURL as jest.Mock).mockReturnValueOnce('test-URL');
+    wrapper = shallow(
+      <EventListenerURL eventListener={mockEventListener} namespace="test-namespace" />,
+    );
+    expect(
+      wrapper
+        .find(ClipboardCopy)
+        .first()
+        .props().children,
+    ).toEqual('test-URL');
```

Comment:
Ideally do not dig into another component during tests; we don't need to verify the internals of other components, it'll serve us no good and just lead us to coupling... perhaps simplify this test to the ClipboardCopy component being found.

Reply:
updated test to just check for existence of ClipboardCopy in the render.